server/src/main/java/org/opensearch/action/search/SearchRequestOperationsListener.java

```diff
@@ -32,6 +33,35 @@ void onRequestStart(SearchRequestContext searchRequestContext) {}
 
     void onRequestEnd(SearchPhaseContext context, SearchRequestContext searchRequestContext) {}
 
+    public void setEnabled(boolean enabled) {
+        if (enabled) {
+            register();
+        } else {
+            deregister();
+        }
+    }
+
+
+    /**
+     * Handler function to register this listener to certain components
+     * This function will be called when the listener is enabled.
+     */
+    protected void register() {
+        if (this.searchRequestListenerManager != null) {
+            this.searchRequestListenerManager.addListener(this);
+        }
+    }
+
+    /**
+     * Handler function to deregister this listener from certain components
+     * This function will be called when the listener is disabled.
+     */
+    protected void deregister() {
+        if (this.searchRequestListenerManager != null) {
+            this.searchRequestListenerManager.removeListener(this);
+        }
+    }
```

Comment:
Should these methods *ever* be called from outside of `setEnabled`? If not, they should be private and final.